benchmarks/benchmarks/bench_function_base.py

```diff
@@ -95,47 +95,159 @@ def time_select_larger(self):
         np.select(self.cond_large, ([self.d, self.e] * 10))
 
 
+class SortGenerator(object):
+    # The size of the unsorted area in the "random unsorted area"
+    # benchmarks
+    AREA_SIZE = 10
+    # The size of the "partially ordered" sub-arrays
+    BUBBLE_SIZE = 10
+
+    @staticmethod
+    def _random(size, dtype):
+        arr = np.arange(size, dtype=dtype)
+        np.random.shuffle(arr)
+        return arr
+    
+    @staticmethod
+    def _ordered(size, dtype):
+        return np.arange(size, dtype=dtype)
+
+    @staticmethod
+    def _reversed(size, dtype):
+        return np.arange(size-1, -1, -1, dtype=dtype)
+
+    @staticmethod
+    def _uniform(size, dtype):
+        return np.ones(size, dtype=dtype)
+
+    @staticmethod
+    def _type_swapped_pair(size, dtype, swap_frac):
+        a = np.arange(size, dtype=dtype)
+        for _ in range(int(size * swap_frac)):
+            x, y = np.random.randint(0, size, 2)
+            a[x], a[y] = a[y], a[x]
+        return a
+
+    @staticmethod
+    def _type_sorted_block(size, dtype, block_size):
+        a = np.arange(size, dtype=dtype)
+        b = []
+        if size < block_size:
+            return a
+        block_num = size // block_size
+        for i in range(block_num):
+            b.extend(a[i::block_num])
+        return np.array(b)
+
+    @staticmethod
+    def _type_random_unsorted_area(size, dtype, frac, area_num):
+        area_num = int(area_num)
+        a = np.arange(size, dtype=dtype)
+        unsorted_len = int(size * frac / area_num)
+        for _ in range(area_num):
+            start = np.random.randint(size-unsorted_len)
+            end = start + unsorted_len
+            np.random.shuffle(a[start:end])
+        return a
+
+    @classmethod
+    def get_array(cls, array_type, size, dtype):
+        import re
+        import functools
+        token_specification = [
+            (r'random',
+                lambda match: cls._random(size, dtype)),
+            (r'ordered',
+                lambda match: cls._ordered(size, dtype)),
+            (r'reversed',
+                lambda match: cls._reversed(size, dtype)),
+            (r'uniform',
+                lambda match: cls._uniform(size, dtype)),
+            (r'sorted\_block\_([0-9]+)',
```

Comment:
I don't think `\_` is special in regex, why not just `_`?

Reply:
No more regex anymore.